examples/plotting/file/line_toggle.py

```diff
@@ -0,0 +1,40 @@
+"""
+Example demonstrating the picking of line objects. The transparency of
+the lines is toggled when clicked.
+"""
+
+import numpy as np
+from bokeh.plotting import output_file, show, figure, Figure
+from bokeh.models import TapTool, CustomJS
+
+# Normally, the use of the tap tool causes Bokehjs to "select" a line
+# when it is clicked on, which is not what we want in this example. To
+# deal with this, we modify the "selected" object to make it look as
+# if no glyph was clicked.
+
+code = """
+d0 = cb_obj.get("selected")["0d"];
+if (d0.glyph) {
+    var alpha = d0.glyph.visuals.line.alpha.value();
+    var color = d0.glyph.visuals.line.color.value();
+    console.log('Toggled line', d0.glyph.id, color);
+    d0.glyph.visuals.line.alpha.fixed_value = (alpha == 1) ? 0.5 : 1.0;
+    // Make selected/unselected lines have normal appearance
+    d0.glyph['renderer'].selection_glyph = null;
+    d0.glyph['renderer'].nonselection_glyph = null;
```

Comment:
The use case is that we want to detect clicks on lines, and then take action based on that. We need code in Python to run, to determine what line was clicked, and based on that, we change the alpha of *several* lines. The UX that we want is very different from Bokeh's standard behavior (where each click means "select this"), so we need to override it. To be honest, it feels a bit awkward to me that Bokeh has such a buildin response to line clicks. It is a response that is common, but in many situations it is not what you want, in chich case it needs to be overridden.

Note that `d0 = cb_obj.get("selected")["0d"]`, so nulling `d0.glyph` modifies the selection object, so to whatever processes the selection after this, it looks like no line was clicked. IMO this is a pretty ok way to prevent Bokeh's internals from taking action on the line click (because `d0.glyph` *is* `null` when no line is clicked). But if you prefer to leave out the example, I can remove it.